target/linux/mpc85xx/image/p1020.mk

```diff
@@ -1,17 +1,27 @@
 define Device/aerohive_hiveap-330
   DEVICE_VENDOR := Aerohive
   DEVICE_MODEL := HiveAP-330
-  DEVICE_PACKAGES := kmod-tpm-i2c-atmel
+  DEVICE_PACKAGES := kmod-tpm-i2c-atmel kmod-mtd-rw uboot-envtools
   BLOCKSIZE := 128k
-  KERNEL := kernel-bin | gzip | uImage gzip
-  KERNEL_SIZE := 8m
-  KERNEL_INITRAMFS := copy-file $(KDIR)/vmlinux-initramfs | uImage none
+  KERNEL := kernel-bin | uImage none
+  KERNEL_INITRAMFS := kernel-bin | uImage none
+  KERNEL_SIZE := 16m
   IMAGES := fdt.bin sysupgrade.bin
   IMAGE/fdt.bin := append-dtb
-  IMAGE/sysupgrade.bin := append-dtb | pad-to 256k | check-size 256k | \
-	append-uImage-fakehdr ramdisk | pad-to 256k | check-size 512k | \
-	append-rootfs | pad-rootfs $$(BLOCKSIZE) | pad-to 41216k | check-size 41216k | \
-	append-kernel | append-metadata
+  IMAGE/sysupgrade.bin := append-dtb | pad-to 256k | append-kernel | \
+	append-rootfs | pad-rootfs | append-metadata
+  DEVICE_COMPAT_VERSION := 2.0
```

Comment:
AFAIK this will also need an entry into base-files/etc/uci-defaults/05_fix-compat-version for that device.
(the mpc85xx target doesn't have one. If you need a reference: see ath79/generic for an example)

This is all needed to let future sysupgrades "know" that 2.0 is already installed.

Reply:
Thanks for the heads up, @chunkeey! I missed this.

I want to make sure users don't forget to run the patching steps outlined in this commit when they are flashing from stock; since there is to TOH page on the wiki for the HiveAP 330, I fear users might not know about this change (I personally use the original commit for instructions to flash the HiveAP 330).

I could use `05_fix-compat-version` to verify the string `owrt_boot` is available in the environment and in the u-boot binary, and only then set the compat version. Would something like this be appropriate?:

```diff
new file   target/linux/mpc85xx/base-files/etc/uci-defaults/05_fix-compat-version
@@ -0,0 +1,18 @@
+. /lib/functions.sh
+. /lib/functions/system.sh
+
+case "$(board_name)" in
+    aerohive,hiveap-330)
+        if [ "$(uci get system.@system[0].compat_version="2.0")" -ne "2.0" ] ; then
+            uboot_part="$(find_mtd_chardev "u-boot")";
+            uboot_env_part="$(find_mtd_chardev "u-boot-env")";
+            if strings "$uboot_part" | grep -m1 -q owrt_boot &&
+                strings "$uboot_env_part" | grep -m1 -q owrt_boot; then
+                    uci set system.@system[0].compat_version="2.0"
+                    uci commit system
+            fi
+        fi
+        ;;
+esac
+
+exit 0
```

Reply:
I wasn't sure at first if the above would be allowed, but after some thought, I think it's a great convenience. I was worried it would be too heavy to run, but it only runs once after wiping configuration thanks to the compat_version check.

Reply:
Added in force-push 684c85c454.

Reply:
Fixed this mistake in force-push ef424384da.

> ```diff
> +        if [ "$(uci get system.@system[0].compat_version="2.0")" -ne "2.0" ] ; then
> ```

Reply:
the "05_fix-compat-version" file does not need any checks, just set the compat_version="2.0".

(this should cover the "sysupgrade -n " case for people that already had coverted.)

Reply:
My concern is that users who have not already converted will not be told they need to convert (or how to do so) if they are booting from an initramfs, as one would be when flashing from stock.

Reply:
Hey @chunkeey: I'm hoping you find the forced-check reasonable. My check in `05_fix-compat-version` does cover `sysupgrade -n` as well as users performing a sysupgrade from within an initramfs.